tests/dispatchers/sync.test.ts

```diff
@@ -0,0 +1,179 @@
+import * as fs from 'fs/promises';
+import { handler } from '../../src/commands/dispatchers/sync';
+import * as dispatcherRepository from '../../src/repositories/dispatchers';
+import { generateMailAction, generateTaskAction } from '../__helpers__/actions';
+import { generateDispatcher, generateMinimalDispatcher } from '../__helpers__/dispatchers';
+
+describe('Dispatchers - Sync', () => {
+  let updateDispatcherSpy;
+  let updateActionSpy;
+  let deleteActionSpy;
+
+  const existingDispatcher = generateDispatcher();
+
+  beforeAll(() => {
+    // TODO: Move these somewhere else?
+    jest.spyOn(dispatcherRepository, 'findAll')
+      .mockImplementation(() => Promise.resolve([existingDispatcher, generateDispatcher()]));
+
+    updateDispatcherSpy = jest.spyOn(dispatcherRepository, 'update')
+      .mockImplementation(() => Promise.resolve({ affectedRecords: 1 }));
+
+    updateActionSpy = jest.spyOn(dispatcherRepository, 'updateAction')
+      .mockImplementation(() => Promise.resolve({ affectedRecords: 1 }));
+
+    deleteActionSpy = jest.spyOn(dispatcherRepository, 'deleteAction')
+      .mockImplementationOnce(() => Promise.resolve({ affectedRecords: 1 }));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Throws for a Dispatcher without a name', async () => {
+    const dispatcher = generateDispatcher({
+      name: undefined,
+    });
+
+    jest.spyOn(fs, 'readFile')
+      .mockImplementationOnce(() => Promise.resolve(JSON.stringify([dispatcher])));
+
+    const error = await handler({ sdk: undefined, file: '' })
+      .catch(e => e);
+
+    expect(error.message).toBe('Dispatcher name is a required field');
+  });
+
+  it('Throws for an Action without a name', async () => {
+    const dispatcher = generateDispatcher({
+      actions: [generateTaskAction({ name: undefined })],
+    });
+
+    jest.spyOn(fs, 'readFile')
+      .mockImplementationOnce(() => Promise.resolve(JSON.stringify([dispatcher])));
+
+    const error = await handler({ sdk: undefined, file: '' })
+      .catch(e => e);
+
+    expect(error.message).toBe('Action name is a required field');
+  });
+
+  it('Creates a Dispatcher with all fields set', async () => {
+    const dispatcher = generateDispatcher();
+
+    jest.spyOn(fs, 'readFile')
+      .mockImplementationOnce(() => Promise.resolve(JSON.stringify([dispatcher])));
+
+    const createDispatcherSpy = jest.spyOn(dispatcherRepository, 'create')
+      .mockImplementationOnce(() => Promise.resolve(dispatcher));
+
+    await handler({ sdk: undefined, file: '' });
+
+    expect(createDispatcherSpy).toHaveBeenCalledTimes(1);
+    expect(createDispatcherSpy).toHaveBeenCalledWith(
+      undefined,
+      expect.objectContaining({
+        name: dispatcher.name,
+        tags: [dispatcher.tags[0], dispatcher.tags[1], 'EXH_CLI_MANAGED'],
```

Comment:
(minor) Can should we use the same variable in the test files for 'EXH_CLI_MANAGED' or at least a variable?

Reply:
I'm torn with this one.... logically yes... but on the other hand, if someone changes it for whatever reasons these tests will fail as they should 🤔